ddtrace/llmobs/_utils.py

```diff
@@ -175,7 +176,7 @@ def _get_ml_app(span: Span) -> str:
         if ml_app is not None:
             return ml_app
         llmobs_parent = _get_nearest_llmobs_ancestor(llmobs_parent)
-    return ml_app or config._llmobs_ml_app or "unknown-ml-app"
+    return ml_app or config._llmobs_ml_app or span.context._meta.get(PROPAGATED_ML_APP_KEY)
```

Comment:
it miiight make more sense to have the propagated ml app have the highest priority.

tentative precedence:

1. ml_app on the span
2. propagated ml_app
3. global ml_app